### PR TITLE
Misc small changes

### PR DESCRIPTION
**Pull request by monoidal**

### CODE REVIEW

File: compiler/basicTypes/Multiplicity.hs [hunk 1]
```diff
@@ -197,3 +198,11 @@ submultMaybe r1 r2 = go r1 r2
...
 --    go (MultThing t) (MultThing t') = Unknown
     go _     _     = Unknown
+
+multThingList :: Multable t => GMult t -> [t]
+multThingList Zero = []
+multThingList One = []
+multThingList Omega = []
+multThingList (MultThing t) = [t]
+multThingList (MultAdd x y) = multThingList x ++ multThingList y
+multThingList (MultMul x y) = multThingList x ++ multThingList y
```
Comment (aspiwack) [inline compiler/basicTypes/Multiplicity.hs]: Why a list rather than a traverse-like thing?

File: testsuite/tests/ghci/scripts/T8959b.stderr [hunk 2]
```diff
@@ -12,5 +12,5 @@ T8959b.hs:8:7: error:
 T8959b.hs:10:7: error:
     • Couldn't match expected type ‘(∀ a2. a2 → a2) → a1’
                   with actual type ‘()’
-    • In the expression: () ∷ (∀ a. a -> a) -> a
-      In an equation for ‘baz’: baz = () ∷ (∀ a. a -> a) -> a
+    • In the expression: () ∷ (∀ a. a → a) → a
+      In an equation for ‘baz’: baz = () ∷ (∀ a. a → a) → a
```
Comment (aspiwack) [inline testsuite/tests/ghci/scripts/T8959b.stderr]: Is this expected?

File: testsuite/tests/linear/should_fail/LinearPartialSig.hs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{-# LANGUAGE LinearTypes #-}
+module LinearPartialSig where
+
+f :: a -->.(_) a
+f x = x
```
Comment (aspiwack) [inline testsuite/tests/linear/should_fail/LinearPartialSig.hs]: What are we testing in this test? Can we document?

File: testsuite/tests/ghci/scripts/T8959b.stderr (hunk 2 above)
Comment (monoidal) [reply testsuite/tests/ghci/scripts/T8959b.stderr]: Yes, I think this should be the expected behavior and this is a bug in master. The test uses -fprint-unicode-syntax.

File: testsuite/tests/linear/should_fail/LinearPartialSig.hs (hunk 3 above)
Comment (monoidal) [reply testsuite/tests/linear/should_fail/LinearPartialSig.hs]: Before ef4fa3397d, this caused an internal error. I added a one line description to the .hs file about the expected result but the motivation is that ghc shouldn't panic.

File: testsuite/tests/ghci/scripts/T8959b.stderr (hunk 2 above)
Comment (monoidal) [reply testsuite/tests/ghci/scripts/T8959b.stderr]: Submitted as https://phabricator.haskell.org/D5439.

**Pull request merged**